src/ima-state-machine.js

```diff
@@ -267,7 +267,7 @@ function onAdCompleted(options: Object, adEvent: any): void {
  */
 function onAdsCompleted(options: Object, adEvent: any): void {
   this.logger.debug(options.transition.toUpperCase());
-  if (this.playOnMainVideoTag() && this._contentComplete) {
+  if (this.playOnMainVideoTag() && this._contentComplete && !this.config.playAdsWithMSE) {
```

Comment:
I think the config name is problematic as it is out of context.
Let's change it to IMA context `reloadSourceOnComplete`